### ISSUE TITLE
Email configuration in realm settings is overriding user_entity entry in database

### ISSUE THREAD

**Issue by radek2s** [keycloak/keycloak#15570]
### Area

core

### Describe the bug

While creating a realm configuration as realm admin user after modification and verification of email settings I was not able to login to admin-console. 

I've noticed that saving this configuration was causing the replacement of current user entry in database to data provided in modal dialog.

### Version

19.02

### Expected behavior

Validation of email configuration  should not change anything in user entry.

## Anonymized example of email configuration 
In this case as Keycloak admin in dedicated realm I wanted to configure SMTP server to send all notifications. 
I've imported this realm data from JSON file but without SMTP server password. 
Additionaly in this configuration I've got in this dedicated realm a private client ("app-backend") that has authorized access and service account enabled.  

Before saving a configuration:
```
select * from user_entity;
```

id | email |  email_constraint | email_verified | enabled | federation_link | first_name | last_name | realm_id | username | created_timestamp | service_account_client_link | not_before 
--- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | ---
 62737ec9-75fa-4389-8105-6cb15c975ab1 |       | ee87af50-d01c-4746-b5c1-e005f3b72155 | f              | t       |                 |            |           | a459b13d-a86e-4057-b55e-5841ba8ce942 | nivadmin                        |     1668769717564 |                                      |          0
 c52aec7a-30e7-4400-9a23-ba38cb0c238e |       | 8bc43791-2016-4be1-aaa2-0fde6eceb853 | f              | t       |                 |            |           | 76b4bf48-5f08-47b6-9103-4295e406b2b8 | service-account-app-backend |     1668769741228 | d2beb3aa-6fe3-4f21-8f0b-a2966a8b3642 |          0

After saving:
```
select * from user_entity;
```
id | email |  email_constraint | email_verified | enabled | federation_link | first_name | last_name | realm_id | username | created_timestamp | service_account_client_link | not_before 
--- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | ---
c52aec7a-30e7-4400-9a23-ba38cb0c238e |                 | 8bc43791-2016-4be1-aaa2-0fde6eceb853 | f              | t       |                 |            |           | 76b4bf48-5f08-47b6-9103-4295e406b2b8 | service-account-app-backend |     1668769741228 | d2beb3aa-6fe3-4f21-8f0b-a2966a8b3642 |          0
 62737ec9-75fa-4389-8105-6cb15c975ab1 | anyone@mail.com | anyone@mail.com                      | f              | t       |                 |            |           | a459b13d-a86e-4057-b55e-5841ba8ce942 | anyone@mail.com                 |     1668769717564 |                                      |          0

## Result:
Username has been replaced

### How to Reproduce?

Even on blank environment we can observe this situation:
![Zrzut ekranu 2022-11-18 o 12 20 25](https://user-images.githubusercontent.com/22638815/202696428-785e27a0-f7f4-46ea-bf8a-1dac9af13552.png)
![Zrzut ekranu 2022-11-18 o 12 25 28](https://user-images.githubusercontent.com/22638815/202696603-207f563c-9fe8-4f7e-badb-92946e9feb8b.png)

1. Login as admin to admin-console
2. Change realm email configuration
3. Validate config

**Comment by lexcao** [comment]
Hi, seems duplicated to https://github.com/keycloak/keycloak-ui/issues/3593.
And is fixed on 20.
Could you please check the latest version of the Keycloak?

**Comment by radek2s** [comment]
Yes on version 20.0 it is working correctly, thanks for reapone!

**Comment by mposolda** [comment]
@lexcao @radek2s Closing, Thanks for the confirm!
